plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/DeleteFile.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.delete;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import org.apache.iceberg.FileContent;
+import org.apache.iceberg.FileFormat;
+import org.openjdk.jol.info.ClassLayout;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static io.airlift.slice.SizeOf.SIZE_OF_INT;
+import static io.airlift.slice.SizeOf.estimatedSizeOf;
+import static java.util.Objects.requireNonNull;
+
+public final class DeleteFile
+{
+    private static final long INSTANCE_SIZE = ClassLayout.parseClass(DeleteFile.class).instanceSize();
+
+    private final FileContent content;
+    private final String path;
+    private final FileFormat format;
+    private final long recordCount;
+    private final long fileSizeInBytes;
+    private final List<Integer> equalityFieldIds;
+
+    @SuppressWarnings("TypeMayBeWeakened")
+    public static DeleteFile fromIceberg(org.apache.iceberg.DeleteFile deleteFile)
+    {
+        return new DeleteFile(
+                deleteFile.content(),
+                deleteFile.path().toString(),
+                deleteFile.format(),
+                deleteFile.recordCount(),
+                deleteFile.fileSizeInBytes(),
+                Optional.ofNullable(deleteFile.equalityFieldIds()).orElseGet(ImmutableList::of));
+    }
+
+    @JsonCreator
+    public DeleteFile(
+            @JsonProperty FileContent content,
```

Comment:
is `@JsonProperty` needed or redundant here?

Reply:
You're right, it's redundant. We could remove these everywhere, now that we rely on parameter names.